lib/xboxrt/wchar.c

```diff
@@ -0,0 +1,43 @@
+#include "wchar.h"
+
+#include <stdint.h>
+
+wchar_t *wcscpy(wchar_t *dst, const wchar_t *src) {
+    return wcsncpy(dst, src, SIZE_MAX);
+}
+
+wchar_t *wcsncpy(wchar_t *dst, const wchar_t *src, size_t n) {
+    size_t i = 0;
+    do dst[i] = *src; while (*src++ != L'\0' && i++ < n);
+    return dst;
+}
+
+wchar_t *wcscat(wchar_t *s1, const wchar_t *s2) {
+    return wcsncat(s1, s2, SIZE_MAX);
+}
+
+wchar_t *wcsncat(wchar_t *s1, const wchar_t *s2, size_t n) {
+    wcsncpy(s1 + wcslen(s1), s2, n);
```

Comment:
Just to prove my point: Shouldn't the last parameter be something like `n - wcslen(s1)` ?

Even if this is not your bug, it will be harder to fix them across our codebase. Especially because this poor / old libc isn't tested at all.

I'm really worried about stability of the software that has been compiled with these "frankensteined" versions of nxdk.

*You should not fix this yet; it's just food for thought. If we want to fix it, we should fix it in the other string function, too.*

Reply:
Yes indeed. I was wrong.

*My argument still holds with the other comment below (I hope): errors might propagate*